### ISSUE TITLE
Error in widget.py : AttributeError: 'Event' object has no attribute 'gain'

### ISSUE THREAD

**Issue by DougOne** [ppizarror/pygame-menu#375]
**Environment information**
Describe your environment information, such as:
- SO: Windows 10
- python version: v3.10.1
- pygame version: v2.1.2
- pygame-menu version: 4.2

**Describe the bug**
Hi, 

I just started to design a game in python based on PyGame and Pygame-menu. 

When I try to execute the simple.py example, I get this error that I can't explain to myself: 

```
Hello from the pygame community. https://www.pygame.org/contribute.html
pygame-menu 4.2.0
Traceback (most recent call last):
  File "d:\User\Dougdoug\Projects\reallybasicpong\main.py", line 50, in <module>
    menu.mainloop(surface)
  File "D:\Software developmentPython\lib\site-packages\pygame_menu.py", line 2910, in mainloop
    self.update(pygame.event.get())
  File "D:\Development softwarePythonlibSite-packagespygame_menu.py", line 2439, in update
    selected_widget.update(events):
  File "D:\Development softwarePython\lib\site-packages\pygame_menu\widgets\widget\textinput.py", line 1570, in update
    self._check_mouseover(event, rect)
  File "D:\Software developmentPython\lib\site-packages\pygame_menu\widgets\core\widget.py", line 692, in _check_mouseover
    if event.gain == 1:
AttributeError: 'Event' object has no attribute 'gain'

```

On the other hand, I manage to execute correctly the other examples present on the repo without problems. 

Any idea ? 

Thanking you in advance.

**Comment by ppizarror** [comment]
Weird... I don't know why it is requesting such event. However, it is super easy to fix. I'm push a new version today :)

**Comment by majsterkovic** [comment]
I got exactly the same error.
```
    if event.gain == 1:
AttributeError: 'Event' object has no attribute 'gain'
```

**Comment by ppizarror** [comment]
I've patched it, but sadly I cannot reproduce the error. Usually pygame.ACTIVEEVENT must have gain attribute:

```bash
```

**Comment by ppizarror** [comment]
In #376 I've patched all ACTIVEEVENT gain calls. Hope this solves any other issue

**Comment by DougOne** [comment]
Hi @ppizarror !
Thank you for the fix. 

Unfortunately, when I try to run simple.py again, I get a different but similar error, namely : 

```
Hello from the pygame community. https://www.pygame.org/contribute.html
pygame-menu 4.2.1
Traceback (most recent call last):
  File "d:\Utilisateur\Dougdoug\Projets\reallybasicpong\menu.py", line 50, in <module>
    menu.mainloop(surface)
  File "D:\Logiciel developpement\Python\lib\site-packages\pygame_menu\menu.py", line 2910, in mainloop
    self.update(pygame.event.get())
  File "D:\Logiciel developpement\Python\lib\site-packages\pygame_menu\menu.py", line 2439, in update
    selected_widget.update(events):
  File "D:\Logiciel developpement\Python\lib\site-packages\pygame_menu\widgets\widget\textinput.py", line 1570, in update
    self._check_mouseover(event, rect)
  File "D:\Logiciel developpement\Python\lib\site-packages\pygame_menu\widgets\core\widget.py", line 707, in _check_mouseover
    rect.collidepoint(*event.pos) and menu_enabled:
AttributeError: 'Event' object has no attribute 'pos'
```

Thank you in advance

**Comment by ppizarror** [comment]
I cannot reproduce with tests. Other examples do not have errors, for example multi-input, maze, or solar system? (which are complex)?

**Comment by DougOne** [comment]
Here my results : 
- scroll_menu.py : 🔴 fail (Error : 'Event' object has no attribute 'pos')
- multi_input.py : 🔴 fail (Error : 'Event' object has no attribute 'pos')
- game_selector : ✔️ 
- ui_solar_system.py : 🔴 fail (Error : 'Event' object has no attribute 'pos')

**Comment by ppizarror** [comment]
This is strange. Maybe that is because of a new pygame version in Windows. Tested in macOS/ubuntu and no problems, and then executed the example examples in windows 11 (parallels) and got the errors. I've uploaded a new version (4.2.2)

**Comment by DougOne** [comment]
Seems to work now for all examples. 
Thank you very much !

I close the issue.

**Comment by majsterkovic** [comment]
@ppizarror, it looks like something is still not working. On Windows 11, the window closes immediately with no error on the console.
```
PS C:\Users\majsterkovic\Desktop\reindeer-gingerbread-game> & "C:/Program Files/Python310/python.exe" c:/Users/majsterkovic/Desktop/reindeer-gingerbread-game/main.py
pygame 2.1.2 (SDL 2.0.18, Python 3.10.0)
Hello from the pygame community. https://www.pygame.org/contribute.html
pygame-menu 4.2.2
PS C:\Users\majsterkovic\Desktop\reindeer-gingerbread-game>

**Comment by ppizarror** [comment]
Hi. Can you provide a minimal working example if possible?

**Comment by ppizarror** [comment]
@majsterkovic, did you remove the issue? Have you solved it? If you encounter any error, please create a minimal working example to check it out in a new issue to solve it quickly. Thanks, and happy new year!

-
Pablo

**Comment by ppizarror** [comment]
Also, I've run all tests in a windows machine (intel) and there was no problems. 
![image](https://user-images.githubusercontent.com/12925256/147839269-1d01bb5c-7541-431a-8043-345cf0e63db3.png)

**Comment by majsterkovic** [comment]
It was my own mistake, sorry.

**Comment by hawkerm** [comment]
I was updating an old pygame project and hit this same message. Did this end up being a pygame issue? Is there an issue over there tracking it as well?

**Comment by ppizarror** [comment]
> I was updating an old pygame project and hit this same message. Did this end up being a pygame issue? Is there an issue over there tracking it as well?

Which version are you using?

**Comment by ppizarror** [comment]
> I was updating an old pygame project and hit this same message. Did this end up being a pygame issue? Is there an issue over there tracking it as well?

I guess it depends on the system and version. Some ones do not have the event attribute, thus I added the following conditions:

https://github.com/ppizarror/pygame-menu/blob/7c281e9d4a8de59b2d53e4d8734b0d231967b8be/pygame_menu/menu.py#L2632-L2634

Let me know if it still happening using the latest pygame-menu version.

**Comment by hawkerm** [comment]
@ppizarror sorry for the confusion. I saw that you created a PR to do a check for the property (I applied that same fix in my pygame project.) Though it seems like just a patch, so I was wondering if you filed or were aware of an underlying issue on the pygame repo. I'm not using `pygame-menu` at present in this project, I had just found your issue about this same event missing property that I did in my project.
